component/prometheus/scrape/scrape.go

```diff
@@ -183,7 +184,11 @@ func (c *Component) Run(ctx context.Context) error {
 				jobName = c.args.JobName
 			}
 			c.mut.RUnlock()
-			promTargets := c.componentTargetsToProm(jobName, tgs)
+
+			// NOTE(@tpaschalis) First approach, manually building the
+			// 'clustered' targets implementation every time.
+			ct := discovery.NewDistributedTargets(c.opts.Clusterer.Node, tgs)
+			promTargets := c.componentTargetsToProm(jobName, ct.Get())
```

Comment:
Is this correct? From what I can tell, this runs even when `clustering_enabled` is false for the component, right? That would cause targets to be sharded and never update if the agent is participating in a cluster but the component isn't subscribed to updates.

Reply:
You're right, this was erroneous. I've fixed this by removing the Registration logic and depending on the Extension interface for now and it works as expected.

I'm exploring the approach with the Capabilities/Registration approaches to see how they compare.